### PR TITLE
[doc] Rewrite Peregrine Overview page

### LINKED ISSUE #1340: [doc]: Improve Peregrine overview page
**Describe the request**

The [Peregrine overview][] page looks bare and does not have much information about what Peregrine is.

[peregrine overview]:https://magento-research.github.io/pwa-studio/peregrine/

**Possible solutions**

Improve the content for Peregrine overview page with the current vision for Peregrine and what it provides.

**Screenshots**

n/a

**Please let us know whether this is a new topic or a topic change request:**
 - [ ] New Topic Request (ie. missing entire topic/section)
 - [x] Topic Change Request (ie. spelling, organization)

### PR DESCRIPTION
**Pull request by jcalcaben**
## Description

Rewrite the Peregrine Overview page to reflect current vision of the Peregrine library.

## Related Issue

Closes #1340 

## Verification Steps

1. Navigate to the `pwa-devdocs` directory
2. Lint the modified page: ` npm run lint src/peregrine/index.md`
3. Verify no linting errors associated with the page
4. Build the HTML preview server: `npm run develop`
5. Verify the site is successfully built
6. Navigate to `/peregrine`
7. Verify content is updated and different from currently published site.

## Screenshots / Screen Captures (if appropriate)

n/a

## Checklist:

- [x] I have updated the documentation accordingly, if necessary.
- [ ] I have added tests to cover my changes, if necessary.

### CODE REVIEW

File: pwa-devdocs/src/peregrine/index.md [hunk 1]
```diff
@@ -2,9 +2,49 @@
...
 
-Visit the [`peregrine`] repository to contribute to this project.
+## Custom React hooks
 
-[`peregrine`]: https://github.com/magento-research/pwa-studio/tree/master/packages/peregrine
\ No newline at end of file
+Many of the functions provided by Peregrine are [custom React hooks][].
+This lets them maintain an internal state without relying on an external library, such as Redux.
+
+Peregrine hooks are designed to be flexible, and non-opinionated.
```
Comment (sirugh) [inline pwa-devdocs/src/peregrine/index.md]:
```suggestion
Peregrine hooks are designed to be flexible, and non-opinionated about UI.
```

File: pwa-devdocs/src/peregrine/index.md [hunk 2]
```diff
@@ -2,9 +2,49 @@
...
+They contain code for providing data or behavior logic and do not render content themselves.
+Rendering content, such as HTML elements, is left up to UI components.
```
Comment (sirugh) [inline pwa-devdocs/src/peregrine/index.md]:
```suggestion
Rendering content, such as HTML elements, is left up to UI components, such as those found in `venia-library`.
```

or w/e we are calling it :D

File: pwa-devdocs/src/peregrine/index.md [hunk 3]
```diff
@@ -2,9 +2,49 @@
...
+
+Separating logic and presentation code gives developers more flexibility on how to use PWA Studio components with their own custom code.
+A developer may choose to use a Venia feature that use certain Peregrine hooks with minor visual modifications, or
```
Comment (sirugh) [inline pwa-devdocs/src/peregrine/index.md]:
```suggestion
A developer may choose to use a Venia feature that uses certain Peregrine hooks with minor visual modifications, or
```

File: pwa-devdocs/src/peregrine/index.md [hunk 4]
```diff
@@ -2,9 +2,49 @@
...
+they can use those same Peregrine hooks to develop their own feature with a different UI.
+
+### Return signatures
+
+Peregrine hooks follow the same pattern as [`useState()`][] and [`useReducer()`][] of returning an array instead of an object.
```
Comment (sirugh) [inline pwa-devdocs/src/peregrine/index.md]: Not necessarily true. In fact `useWindowSize` returns an object. Maybe just note that signatures can vary and can return an array with "state and api", similar to `useState`, or they may return an object. Maybe provide a reference to [React's docs](https://reactjs.org/docs/hooks-custom.html) for writing custom hooks.

File: pwa-devdocs/src/peregrine/index.md (hunk 2 above)
Comment (supernova-at) [inline pwa-devdocs/src/peregrine/index.md]:
```suggestion
Rendering content is left up to UI components.
```

File: pwa-devdocs/src/peregrine/index.md [hunk 5]
```diff
@@ -2,9 +2,49 @@
...
+This lets you specify the variable names of the provided objects when you [destructure][] the array.
+
+Array destructuring example:
+
+```js
+const [myDataObject, myDataObjectApi] = usePeregrineHook();
+```
```
Comment (supernova-at) [inline pwa-devdocs/src/peregrine/index.md]: This [isn't always the case](https://github.com/magento-research/pwa-studio/blob/develop/packages/peregrine/src/hooks/useDropdown.js) and I don't think we want to bind ourselves to this.

We're keeping an eye on consistency but some hooks will necessarily have to deviate from this pattern.

**Comment by dpatil-magento** [comment]
QA Pass.

**Comment by jimbo** [comment]
Consider not using the phrase `UI components`, as that phrase has meaning in Magento that doesn't carry over to PWA Studio.

**Comment by jcalcaben** [comment]
I'll create an issue for a docs-wide audit for the term `UI component`

**Pull request merged**